pkg/networkpolicy/controller.go

```diff
@@ -391,8 +391,8 @@ func (c *Controller) Run(ctx context.Context) error {
 
 		packet, err := parsePacket(*a.Payload)
 		if err != nil {
-			klog.Infof("Can not process packet %d applying default policy (failOpen: %v): %v", *a.PacketID, c.config.FailOpen, err)
-			c.nfq.SetVerdict(*a.PacketID, verdict) //nolint:errcheck
+			klog.Infof("Can not process packet %d, allowing ... : %v", *a.PacketID, err)
+			c.nfq.SetVerdict(*a.PacketID, nfqueue.NfAccept) //nolint:errcheck
```

Comment:
I'm not sure this is really correct

Reply:
this means this packet is not TCP or UDP or SCTP

Reply:
No, it means "a parsing error occurred". I'm not sure we want to say all parse errors should be "accepts".

(I'm not sure we _don't_ want that either...)

Reply:
I see, it can only happens if is not ip or ipv6 or one of those protocols ... but indeed is confusing the way it is now

Reply:
All packets with unknown protocol (not TCP or UDP or SCTP) _must_ be accepted. Unfortunately there is no `v1.ProtocolUnknown` constant, but we can type-cast:
```go
	default:
		//return t, fmt.Errorf("unknown protocol %d", protocol)
		t.proto = v1.Protocol("unknown")
```
but that feels a bit hackish...

Reply:
we don't need to cast to v1.protocol type, we can use a string for that field or create our own enum
The code is this way just by commodity, but if it does not fit the design we should adapt it accordingly

Reply:
Would it be ok to just store the proto as an int in `struct packet`, and let the caller match on "syscall.IPPROTO_*". IMO that's the most flexible.

Reply:
My point was that it doesn't seem right to say that every possible type of error from `parsePacket` should result in the packet being accepted. IMO it would be better to have `parsePacket` return both an error and a verdict, even if that verdict happens to be "accept" for all of the errors it _currently_ catches.

Reply:
After some investigation, I agree: all unrecognized packets shall not be accepted (https://github.com/kubernetes-sigs/kube-network-policies/pull/51#issuecomment-2216969955). However, I think `parsePacket()` should not set a verdict. It should parse the packet and let the caller decide, but then is can't return TCP,UDP,SCTP or an error for everything else.

Reply:
> Would it be ok to just store the proto as an int in `struct packet`, and let the caller match on "syscall.IPPROTO_*". IMO that's the most flexible.

that sounds like the best idea

Reply:
I have experimented some with this, and I think the best is to keep packet.proto as v1.Protocol, and add a new "ipproto int" in the packet structure.

An IPPROTO_ROUTING extension header, or extension header out-of-order will return a
```
var ErrorExtensionHeader = fmt.Errorf("unsupported extension header")
```